tiktoken_ext/openai_public.py

```diff
@@ -6,6 +6,11 @@
 FIM_SUFFIX = "<|fim_suffix|>"
 ENDOFPROMPT = "<|endofprompt|>"
 
+# The pattern in the original GPT-2 release is:
+# r"""'s|'t|'re|'ve|'m|'ll|'d| ?[\p{L}]+| ?[\p{N}]+| ?[^\s\p{L}\p{N}]+|\s+(?!\S)|\s+"""
+# This is equivalent, but executes faster:
+_legacy_splitter_regex = r"""'(?:[sdmt]|ll|ve|re)| ?\p{L}++| ?\p{N}++| ?[^\s\p{L}\p{N}]++|\s+(?!\S)|\s++"""
```

Comment:
The whitespaces can't be possessive (it needs to step back when encountering a non-shitespace), I have tried atomic grouping here, tried possessives with positive lookeahed, but couldn't make it work.
So all-whitespace humonguous inputs will still fail, everything else is likely fixed